source/plugins/ruby/in_kubestate_hpa.rb

```diff
@@ -132,6 +133,18 @@ def enumerate
           $log.info("in_kubestate_hpa::hpaCount= #{hpaCount}")
           #ApplicationInsightsUtility.sendMetricTelemetry("HPACount", @hpaCount, {})
         end
+
+        # Adding telemetry to send kubestateHPA telemetry every 5 minutes
+        timeDifference = (DateTime.now.to_time.to_i - @@kubeStateHPATelemetryTimeTracker).abs
+        timeDifferenceInMinutes = timeDifference / 60
+        if (timeDifferenceInMinutes >= 5)
+          telemetryFlush = true
+        end
+
+        if telemetryFlush
+          ApplicationInsightsUtility.sendCustomEvent("KubeStateHPAHeartBeatEvent", {})
```

Comment:
same here as well. telemetryFlush needs to set to false to avoid event being emitted for every iteration.